pkg/controller/common/license/model.go

```diff
@@ -43,6 +43,12 @@ type LicenseSpec struct {
 	ClusterLicenses    []ElasticsearchLicense `json:"cluster_licenses"`
 }
 
+// EnterpriseLicenseTypeOrder license types mapped to ints in increasing order of feature sets for sorting purposes.
+var EnterpriseLicenseTypeOrder = map[OperatorLicenseType]int{
```

Comment:
I always struggle with the iota semantics but I wonder if that might be e better fit here

Reply:
I do see that we use a map here for ES licenses though
https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/elasticsearch/client/model.go#L236

Reply:
Like you I'm not a big fan of the iota semantics. 

The stack license order is defined in the same way.

https://github.com/elastic/cloud-on-k8s/blob/d212d415a0651066ad2f6fd506929c0e3fa976e8/pkg/controller/elasticsearch/client/model.go#L249-L255